### PR TITLE
Update user input per question error when value changes.

### PR DESCRIPTION
**Pull request by kuasha420**
## Summary

Addresses issue:

- #7543

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/components/settings/SettingsCardKeyMetrics.js [hunk 1]
```diff
@@ -56,22 +53,6 @@ export default function SettingsCardKeyMetrics() {
 		}
 	}, [ isUserInputCompleted, gaEventCategory ] );
 
-	const { navigateTo } = useDispatch( CORE_LOCATION );
-	const goTo = ( questionIndex = 1 ) => {
```
Comment (kuasha420) [inline assets/js/components/settings/SettingsCardKeyMetrics.js:60]: Relic of User Input V1, no longer needed

File: assets/js/components/settings/SettingsCardKeyMetrics.js [hunk 2]
```diff
@@ -89,13 +70,7 @@ export default function SettingsCardKeyMetrics() {
 					{ isUserInputCompleted && (
 						<Row>
 							<Cell size={ 12 }>
-								<UserInputPreview
-									goTo={ goTo }
-									noHeader
-									noFooter
-									settingsView
-									showIndividualCTAs
```
Comment (kuasha420) [inline assets/js/components/settings/SettingsCardKeyMetrics.js:97]: In two instances where the `UserInputPreview` is used, this 4 prop means the same thing, only one is sufficient.

File: assets/js/components/user-input/UserInputPreview.js [hunk 3]
```diff
@@ -86,13 +77,12 @@ export default function UserInputPreview( props ) {
...
 	} = getUserInputAnswers();
+
 	const [ page ] = useQueryArg( 'page' );
 
-	const [ errorMessages, setErrorMessages ] = useState( {
-		[ USER_INPUT_QUESTIONS_GOALS ]: null,
-		[ USER_INPUT_QUESTIONS_PURPOSE ]: null,
-		[ USER_INPUT_QUESTION_POST_FREQUENCY ]: null,
-	} );
+	const hasError = USER_INPUT_QUESTIONS_LIST.some( ( slug ) =>
```
Comment (kuasha420) [inline assets/js/components/user-input/UserInputPreview.js:83]: If there is error in any question, the save button should be disabled, not silently fail.

File: assets/js/components/user-input/UserInputPreview.js [hunk 4]
```diff
@@ -115,33 +105,6 @@ export default function UserInputPreview( props ) {
...
-	};
-
-	const onSaveClick = () => {
-		const newErrorMessages = updateErrorMessages();
-
-		const hasErrors = Object.values( newErrorMessages ).some( Boolean );
-		if ( ! hasErrors ) {
-			submitChanges();
-		}
-	};
```
Comment (kuasha420) [inline assets/js/components/user-input/UserInputPreview.js:143]: As per above, this dance is no longer needed, as the save button will be disabled if there is any error.

File: assets/js/components/user-input/UserInputPreview.js [hunk 5]
```diff
@@ -250,11 +203,7 @@ export default function UserInputPreview( props ) {
 
 UserInputPreview.propTypes = {
 	submitChanges: PropTypes.func,
-	noFooter: PropTypes.bool,
 	goBack: PropTypes.func,
-	redirectURL: PropTypes.string,
-	errors: PropTypes.object,
-	noHeader: PropTypes.bool,
+	error: PropTypes.object,
```
Comment (kuasha420) [inline assets/js/components/user-input/UserInputPreview.js:207]: This error is not the error about answers, but the potential error for `saveUserInputSettings` action which is passed from the `UserInputQuestionnaire`.

File: assets/js/components/user-input/UserInputPreviewGroup.js [hunk 6]
```diff
@@ -193,20 +189,18 @@ export default function UserInputPreviewGroup( {
...
 						slug={ slug }
 						max={ USER_INPUT_MAX_ANSWERS[ slug ] }
 						options={ options }
 						alignLeftOptions
 					/>
-					{ errorMessage && (
-						<p className="googlesitekit-error-text">
-							{ errorMessage }
-						</p>
+					{ error && (
```
Comment (kuasha420) [inline assets/js/components/user-input/UserInputPreviewGroup.js:200]: This error is updated on value change (more like on every re-render, but close enough!)

**Review by kuasha420 — COMMENTED** [review]
Some some clarification for potential code reviewer, feedback welcome!

File: assets/js/components/user-input/UserInputPreview.js (hunk 3 above)
Comment (hussain-t) [reply assets/js/components/user-input/UserInputPreview.js:83]: Makes sense 👍

**Review by hussain-t — APPROVED** [review]
Thanks, @kuasha420. It's great to see such refactoring and removing unnecessary code. Nice one 👍

**Review by tofumatt — APPROVED** [review]

**Pull request merged**